echopop/live/live_acoustics.py

```diff
@@ -0,0 +1,337 @@
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from ..acoustics import to_linear, ts_length_regression
+from .live_spatial_methods import apply_griddify_definitions, apply_spatial_definitions
+from .sql_methods import query_processed_files, sql_data_exchange
+
+
+# TODO: Documentation
+def configure_transmit_frequency(
+    frequency_values: pd.Series, transmit_settings: dict, current_units: str
+):
+
+    # Extract transmit frequency units defined in configuration file
+    configuration_units = transmit_settings["units"]
+
+    # Transform the units, if necessary
+    # ---- Hz to kHz
+    if current_units == "Hz" and configuration_units == "kHz":
+        return frequency_values * 1e-3
+    # ---- kHz to Hz
+    elif current_units == "kHz" and configuration_units == "Hz":
+        return frequency_values * 1e3
+    # ---- No change
+    else:
+        return frequency_values
+
+
+# TODO: Documentation
+def preprocess_acoustic_data(
+    survey_data: pd.DataFrame, spatial_dict: dict, file_configuration: dict
+) -> pd.DataFrame:
+
+    # Get acoustic processing settings
+    acoustic_analysis_settings = file_configuration["acoustics"]
+    # ---- Extract the fined acoustic frequency
+    transmit_settings = acoustic_analysis_settings["transmit"]
+
+    # Filter the dataset
+    # ---- Configure `frequency_nominal`, if necessary
+    survey_data.loc[:, "frequency_nominal"] = configure_transmit_frequency(
+        survey_data.loc[:, "frequency_nominal"],
+        transmit_settings,
+        acoustic_analysis_settings["dataset_units"]["frequency"],
+    )
+    # ---- Filter out any unused frequency coordinates
+    prc_nasc_df_filtered = (
+        survey_data[survey_data["frequency_nominal"] == transmit_settings["frequency"]]
+        # ---- Drop NaN/NaT values from longitude/latitude/ping_time
+        .dropna(subset=["longitude", "latitude", "ping_time"])
+    )
```

Comment:
I think it will be more efficient to do this selection in `read_acoustic_files` so that what you're doing in this function will be strictly related to spatial stuff (and then the function name can be made more specific also).